examples/iterators.cpp

```diff
@@ -0,0 +1,263 @@
+#include <xtypes/xtypes.hpp>
+#include <xtypes/Module.hpp>
+
+#include <iostream>
+
+using namespace eprosima::xtypes;
+
+int main()
+{
+    std::cout << "--- Iterate string ---" << std::endl;
+    {
+        StringType string;
+        DynamicData str1(string);
+
+        str1.value<std::string>("Hello!");
+        for (ReadableDynamicDataRef&& elem : str1)
+        {
+            std::cout << elem.value<char>();
+        }
+        std::cout << std::endl;
+
+        // Hello! -> Ifmmp"
+        for (WritableDynamicDataRef&& elem : str1)
+        {
+            elem = static_cast<char>(elem.value<char>() + 1);
+        }
+
+        for (ReadableDynamicDataRef&& elem : str1)
+        {
+            std::cout << elem.value<char>();
+        }
+        std::cout << std::endl;
+    }
+
+    std::cout << "--- Iterate array ---" << std::endl;
+    {
+        ArrayType array_type(primitive_type<int32_t>(), 10);
+        DynamicData array(array_type);
+
+        for (int i = 0; i < 10; ++i)
+        {
+            array[i] = 5 * i;
+        }
+
+        int32_t check_sum = 0;
+        for (ReadableDynamicDataRef&& elem : array)
+        {
+            check_sum += elem.value<int32_t>();
```

Comment:
Do you sure you need the `.value<int32>()`? (see ocurrences)

Reply:
Yes, because of the `operator+=` ambiguity.

Reply:
Can we fix it with the addition of these kinds of operators?

Reply:
Spoken in person